hazelcast/src/test/java/com/hazelcast/jet/JobStatusListenerTest.java

```diff
@@ -301,30 +282,40 @@ protected void testListener(Object source, Consumer<Job> test, String... log) {
     }
 
     protected void testLightListener(Object source, BiConsumer<Job, JobStatusLogger> test) {
+        testListener(source, JetService::newLightJob, test);
+    }
+
+    protected void testLightListener(Object source, Consumer<Job> test, String log) {
+        testLightListener(source, (job, listener) -> {
+            test.accept(job);
+            assertEqualsEventually(listener.log, log);
+        });
+    }
+
+    private void testListener(Object source, BiFunction<JetService, Pipeline, Job> submit,
+                              BiConsumer<Job, JobStatusLogger> test) {
         Pipeline p = Pipeline.create();
         (source instanceof BatchSource
                     ? p.readFrom((BatchSource<?>) source)
                     : p.readFrom((StreamSource<?>) source).withoutTimestamps())
                 .writeTo(Sinks.noop());
 
-        Job job = instance.get().getJet().newLightJob(p);
-        JobStatusLogger listener = new JobStatusLogger(job, -1);
+        Job job = submit.apply(instance.get().getJet(), p);
+        JobStatusLogger listener = new JobStatusLogger(job);
         jobIdString = job.getIdString();
         registrationId = listener.registrationId;
         test.accept(job, listener);
     }
 
-    protected void testLightListener(Object source, Consumer<Job> test, String log) {
-        testLightListener(source, (job, listener) -> {
-            test.accept(job);
-            assertIterableEqualsEventually(listener.log, log);
+    @SafeVarargs
+    protected static <T> void assertEqualsEventually(List<T> actual, T... expected) {
+        assertTrueEventually(() -> {
+            List<T> actualCopy = new ArrayList<>(actual);
+            assertEquals("length", actualCopy.size(), expected.length);
+            assertEquals(actualCopy, asList(expected));
         });
     }
 
-    protected static void assertIterableEqualsEventually(Iterable<?> actual, Object... expected) {
-        assertTrueEventually(() -> assertIterableEquals(actual, expected));
-    }
-
     @SafeVarargs
```

Comment:
`assertTailEqualsEventually` seems not to be needed anymore, can be removed (also from EE)

Reply:
It was added because `NOT_RUNNING` → `STARTING` and `STARTING` → `RUNNING` transitions might be missed. I confirmed that this is still the case by trying to remove it.